pages/rest_api/artifacts.md.erb

```diff
@@ -24,6 +24,90 @@ An artifact is a file uploaded by your agent during the execution of a build’s
 </tbody>
 </table>
 
+## Glob path pattern
+
+Glob path patterns are used throughout Buildkite for artifact file paths. We use a [custom globbing function](https://github.com/buildkite/agent/blob/16a9807a91c4dd1388649557108d4092ae4860a4/glob/glob.go) on the backend that expands the patterns available. The path patterns differ slightly when uploading and downloading artifacts. 
+
+### Uploading artifacts
+
+The pattern that you define should be designed to match whole strings rather than substrings. 
+Keep in mind while you're writing your path pattern:
+
+- there are two wildcards available that match non-separator characters:
+  + `*` to match a sequence of characters
+  + `?` to match a single character
+- character ranges surrounded by `[]` support the `^` as a negator
+- special characters can be escaped with `\\`
+- multiple paths are separated with `;`
+- surround the pattern with quotes
+
+The source path of your artifact is the one that will be used at the destination as well. If you just want the item, and not the full path name, make sure you `cd` into the directory containing your artifact before running the upload. When you run:
+
+```bash
+buildkite-agent artifact upload "log/test.log"
+```
+
+Buildkite will show the file as `log/test.log` in the user interface. If you'd rather it be labeled as just `test.log`, then you'll need to change the working directory of the buildkite-agent before running upload like so:
+
+```bash
+cd log
+buildkite-agent artifact upload "test.log"
+```
+
+Some of the common uses for glob paths patterns are:
+
+Uploading a specific file
+
+```bash
+buildkite-agent artifact upload "log/test.log"
+```
+
+Uploading all the jpegs and pngs, in all folders and subfolders
+
+```bash
+buildkite-agent artifact upload "*/**/*.jpg;*/**/*.jpeg;*/**/*.png"
+```
+
+Uploading all the log files in the log folder
+
+```bash
+buildkite-agent artifact upload "log/*.log"
+```
+
+Uploading all the files and folders inside the `coverage` directory
+
+```bash
+buildkite-agent artifact upload "coverage/**/*"
+```
+
+Uploading a file name with special characters (e.g hello??.html)
+
+```bash
+buildkite-agent artifact upload "hello\?\?.html"
+```
+
+### Downloading artifacts
+
+The path patterns for downloading artifacts are a little different, as they are translated into SQL LIKE statements to query our database. To enable this translation, the path pattern only supports text matching and the single wildcard syntax `*`. 
```

Comment:
I'd probably not mention the SQL gear, we may change how we do it in the future:

> The path patterns for downloading artifacts are different to uploads. Buildkite only supports a `*` wildcard character something something.